src/archives/util.cpp

```diff
@@ -52,7 +52,11 @@ static Common::UString findPath(const Common::UString &name, Aurora::FileType ty
 	Common::UString path;
 
 	if (!name.empty())
-		path = TypeMan.addFileType(name, type);
+		// Check if this name does not have a file type ending.
+		if (TypeMan.getFileType(name) == Aurora::kFileTypeNone)
+			path = TypeMan.addFileType(name, type);
+		else
+			path = name;
```

Comment:
This breaks the unherf tool, because Sonic HERF files contain files with multiple extensions. Namely files compressed with the "small"-algorithm.

Like, for example, itm_ace_timmy_0.ncgr.small is now shown/extracted as itm_ace_timmy_0.ncgr, which is wrong.

Reply:
Are .small files the only occasion where this happens? Becauseif so i would simply add a condition like
```cpp
if (TypeMan.getFileType(name) == Aurora::kFileTypeNone || type == Aurora::kFileTypeSMALL)
```

Reply:
Yeah, I don't want such a hacky solution :P

Reply:
No, you should not

Reply:
Wait, *why* is that necessary to begin with?

File archives shouldn't carry an extension in their filenames.

I.e. the real, actual issue is that the `TheWitcherSaveFile` Archive class should strip extensions from the filenames, like we're doing with ZIP files, certain ERFs, NDS, HERFs, OBB, ... (see https://github.com/xoreos/xoreos/blob/master/src/aurora/zipfile.cpp#L61, https://github.com/xoreos/xoreos/blob/master/src/aurora/erffile.cpp#L600, https://github.com/xoreos/xoreos/blob/master/src/aurora/herffile.cpp#L133, https://github.com/xoreos/xoreos/blob/master/src/aurora/ndsrom.cpp#L92, https://github.com/xoreos/xoreos/blob/master/src/aurora/obbfile.cpp#L97)

Reply:
And yes, I probably should have caught that earlier, sorry